cadence/contracts/AxelarGateway.cdc

```diff
@@ -0,0 +1,312 @@
+import AxelarAuthWeighted from "./auth/AxelarAuthWeighted.cdc"
+import Crypto
+
+// Main GateWay contract for audit
+access(all) contract AxelarGateway {
+  access(self) let SELECTOR_APPROVE_CONTRACT_CALL: [UInt8]
+  access(self) let SELECTOR_TRANSFER_OPERATORSHIP: [UInt8]
+  access(all) let PREFIX_APP_CAPABILITY_NAME: String
+
+  access(all) resource CGPCommand {
+    access(all) var commandId: String
+    access(all) var isExecuted: Bool
+
+    destroy() {
+      emit Executed(commandId: self.commandId)
+    }
+
+    init(commandId: String) {
+      self.commandId = commandId
+      self.isExecuted = false
+    }
+  }
+
+  /**********\
+  |* Events *|
+  \**********/
+  access(all) event ContractCall(
+    sender: Address,
+    destinationChain: String,
+    destinationContractAddress: String,
+    payloadHash: String,
+    payload: [UInt8]
+  )
+
+  access(all) event Executed(commandId: String)
```

Comment:
How are command ids determined? By caller or is there some shared schema? I'm wondering how collisions are prevented in this shared namespace and also how we could make this event more useful to the interested party - i.e. the individual who sourced the command.

Reply:
Command ids are determined in the Axelar Network. I believe it is generated from the transaction hash and payload hash, I could be wrong. However, I confirmed with Axelar team that the command ids will always be unique, so there shouldn't be collisions.